### PR TITLE
Improve documentation for NUM_CPU

### LINKED ISSUE #8403: Rewrite documentation on NUM_CPU
The docs for NUM_CPU is slightly bad (doesn't really try to explain what it is) and is outdated with respect to what it checks (for Torque). Should mention its relation to Eclipse.

https://fmu-docs.equinor.com/docs/ert/reference/configuration/keywords.html#num-cpu

### PR DESCRIPTION
**Pull request by berland**
Don't mention any longer requirements for Torque, which are now deprecated.

**Issue**
Resolves #8403 

**Approach**
✍🏻 

- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Added appropriate release note label
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: docs/reference/configuration/keywords.rst [hunk 1]
```diff
@@ -257,25 +254,27 @@ NUM_CPU
...
+this is implicitly set. If you need to override, use ``NUM_CPU`` explicitly.
 
-For TORQUE, it literally is a check that ``NUM_CPU`` is larger than the amount
-of resources TORQUE wants to allocate (number_of_nodes \* cpus_per_node). See
-:ref:`NUM_NODES|NUM_CPUS_PER_NODE <torque_nodes_cpus>` for details.
+This number should be regarded as informational to the queueing system and is
+likely to affect scheduling, in that a realization will not start until
+sufficient CPU resources are assumed available. It will not guarantee CPU
+exclusiveness on the compute node. If set incorrectly, there is no guarantee
+that the realization will fail.
```
Comment (jonathan-eq) [inline docs/reference/configuration/keywords.rst:273]:
```suggestion
exclusiveness on the compute node. If set incorrectly, there is no guarantee
that the realization will run.
```
Is this correct?

**Review by jonathan-eq — CHANGES_REQUESTED** [review]
Very nice!
See minor question 📝

File: docs/reference/configuration/keywords.rst (hunk 1 above)
Comment (berland) [reply docs/reference/configuration/keywords.rst:273]: Nope. You are lucky if you are able to fail your realization by specifying wrong cpu count. We had only a few failures when num_cpu was missing altogheter in the lsf_driver and hundreds of multi-cpu Eclipse runs were running and all of them said to the LSF system what "I only need 1 cpu".

**Review by jonathan-eq — APPROVED** [review]
LGTM!

**Comment by berland** [comment]
@sondreso any comment?

**Review by sondreso — APPROVED** [review]
Except for the small comment from @jonathan-eq, this  LGTM!

**Pull request merged**